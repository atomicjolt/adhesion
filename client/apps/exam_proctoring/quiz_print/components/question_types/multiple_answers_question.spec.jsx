import React from 'react';
import { shallow } from 'enzyme';
import MultipleChoice from './multiple_answers_question';

describe('Multiple Choice', () => {
  let result;
  let props;

  beforeEach(() => {
    props = {
      answers: [
        {
          id: 1,
          text: 'grass'
        },
        {
          id: 2,
          text: 'potatoes'
        },
        {
          id: 3,
          text: 'trees'
        },
        {
          id: 4,
          text: 'bats'
        },
        {
          id: 5,
          text: 'fear'
        },
      ],
      question_text: '<p>What things are green</p>',
      id: 7,
    };
    result = shallow(<MultipleChoice {...props} />);
  });

  it('matches the snapshot', () => {
    expect(result).toMatchSnapshot();
  });

  it('renders radio buttons for the options', () => {
    const boxes = result.find('input');
    expect(boxes.length).toBe(5);
    expect(boxes.at(0).props().type).toEqual('checkbox');
  });
});
