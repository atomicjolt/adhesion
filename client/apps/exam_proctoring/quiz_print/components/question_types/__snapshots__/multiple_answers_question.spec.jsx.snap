// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Multiple Choice matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <multipleAnswerQuestion
    answers={
        Array [
            Object {
              "id": 1,
              "text": "grass",
            },
            Object {
              "id": 2,
              "text": "potatoes",
            },
            Object {
              "id": 3,
              "text": "trees",
            },
            Object {
              "id": 4,
              "text": "bats",
            },
            Object {
              "id": 5,
              "text": "fear",
            },
          ]
    }
    id={7}
    question_text="<p>What things are green</p>"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <essayQuestion
          text="<p>What things are green</p>"
/>,
        Array [
          <div>
            <hr />
            <input
                        disabled={true}
                        type="checkbox"
            />
             
            grass
</div>,
          <div>
            <hr />
            <input
                        disabled={true}
                        type="checkbox"
            />
             
            potatoes
</div>,
          <div>
            <hr />
            <input
                        disabled={true}
                        type="checkbox"
            />
             
            trees
</div>,
          <div>
            <hr />
            <input
                        disabled={true}
                        type="checkbox"
            />
             
            bats
</div>,
          <div>
            <hr />
            <input
                        disabled={true}
                        type="checkbox"
            />
             
            fear
</div>,
        ],
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "text": "<p>What things are green</p>",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "question7_answer1",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <hr />,
            <input
              disabled={true}
              type="checkbox"
/>,
            " ",
            "grass",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "hr",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "disabled": true,
              "type": "checkbox",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          " ",
          "grass",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "question7_answer2",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <hr />,
            <input
              disabled={true}
              type="checkbox"
/>,
            " ",
            "potatoes",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "hr",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "disabled": true,
              "type": "checkbox",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          " ",
          "potatoes",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "question7_answer3",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <hr />,
            <input
              disabled={true}
              type="checkbox"
/>,
            " ",
            "trees",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "hr",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "disabled": true,
              "type": "checkbox",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          " ",
          "trees",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "question7_answer4",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <hr />,
            <input
              disabled={true}
              type="checkbox"
/>,
            " ",
            "bats",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "hr",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "disabled": true,
              "type": "checkbox",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          " ",
          "bats",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "question7_answer5",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <hr />,
            <input
              disabled={true}
              type="checkbox"
/>,
            " ",
            "fear",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "hr",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "disabled": true,
              "type": "checkbox",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          " ",
          "fear",
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <essayQuestion
            text="<p>What things are green</p>"
/>,
          Array [
            <div>
              <hr />
              <input
                            disabled={true}
                            type="checkbox"
              />
               
              grass
</div>,
            <div>
              <hr />
              <input
                            disabled={true}
                            type="checkbox"
              />
               
              potatoes
</div>,
            <div>
              <hr />
              <input
                            disabled={true}
                            type="checkbox"
              />
               
              trees
</div>,
            <div>
              <hr />
              <input
                            disabled={true}
                            type="checkbox"
              />
               
              bats
</div>,
            <div>
              <hr />
              <input
                            disabled={true}
                            type="checkbox"
              />
               
              fear
</div>,
          ],
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "text": "<p>What things are green</p>",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "question7_answer1",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <hr />,
              <input
                disabled={true}
                type="checkbox"
/>,
              " ",
              "grass",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "hr",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "disabled": true,
                "type": "checkbox",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            " ",
            "grass",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "question7_answer2",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <hr />,
              <input
                disabled={true}
                type="checkbox"
/>,
              " ",
              "potatoes",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "hr",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "disabled": true,
                "type": "checkbox",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            " ",
            "potatoes",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "question7_answer3",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <hr />,
              <input
                disabled={true}
                type="checkbox"
/>,
              " ",
              "trees",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "hr",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "disabled": true,
                "type": "checkbox",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            " ",
            "trees",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "question7_answer4",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <hr />,
              <input
                disabled={true}
                type="checkbox"
/>,
              " ",
              "bats",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "hr",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "disabled": true,
                "type": "checkbox",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            " ",
            "bats",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "question7_answer5",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <hr />,
              <input
                disabled={true}
                type="checkbox"
/>,
              " ",
              "fear",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "hr",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "disabled": true,
                "type": "checkbox",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            " ",
            "fear",
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
